server.go

```diff
@@ -0,0 +1,79 @@
+package baseplate
```

Comment:
Having this be top-level rather than in an inner directory called `baseplate` seems like it'll probably get messy?  Also it means that the package name won't map up with what the importer presents which is kind of annoying (although not exactly _uncommon_).

Reply:
by "messy" do you mean the dependencies it pulled in? that won't have a difference between being top level or subdirectory, because even at top level it won't be pulled into dependency tree unless the top level one is explicitly imported (e.g. `imporet "github.com/reddit/baseplate.go/log"` won't auto pull `github.com/reddit/baseplate.go`).

For the rename importing, yes but that's a minor issue. The alternative is to use `github.com/reddit/baseplate.go/baseplate`, which is arguably too redundant.

Reply:
In this case, I mean `messy` in that now there's `go` code in the top level directory rather than having everything organized into sub-directories (with dedicated doc.go files).  For one file, maybe that's fine but if this top-level package grows, it's going to be messy to have code intermingled with the other non-code files in the top-level directory.

I think either side of the "import" argument is minor either way as you're likely only going to be importing this once in `main.go`, mostly just a point to call out.

Reply:
Yea I understand your concern. I think to address that we just need to be more careful about what to pull into the top level package and what to not pull into it.

I think we should position the top-level package as a "all/most battery included starting point" for service implementations, the stuff they need to do in their main function. With that purpose in mind, do you think it's more acceptable?

Another reason I prefer top level vs. a "baseplate" subdirectory is for the doc rendering. Since this is a starting point, the doc rendered will be a good candidate to be put on to https://pkg.go.dev/mod/github.com/reddit/baseplate.go

Reply:
I think that makes sense, especially from the documentation/onboarding perspective.